plugin/plugins/filesystem/nodes/ipns.go

```diff
@@ -0,0 +1,252 @@
+package fsnodes
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/djdv/p9/p9"
+	files "github.com/ipfs/go-ipfs-files"
+	logging "github.com/ipfs/go-log"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	corepath "github.com/ipfs/interface-go-ipfs-core/path"
+)
+
+// IPNS exposes the IPNS API over a p9.File interface
+// Walk does not expect a namespace, only its path argument
+// e.g. `ipfs.Walk([]string("Qm...", "subdir")` not `ipfs.Walk([]string("ipns", "Qm...", "subdir")`
+type IPNS struct {
```

Comment:
Let's move all this to a new PR.

Reply:
Also, this looks like a lot of duplicate code.

Reply:
Resolving IPNS related comments as the IPNS PR will be separate.